### PR TITLE
feat: Extend `tree` library to support for getting a subtree, iterator over children and conversion back to human-readable form

### PR DESCRIPTION
**Pull request by KirilMihaylov**

### CODE REVIEW

File: packages/tree/src/node.rs [hunk 1]
```diff
@@ -46,13 +51,59 @@ impl<'r, T> NodeRef<'r, T> {
...
+
+pub struct ParentsIter<'r, T> {
+    node: NodeRef<'r, T>,
+}
+
+impl<'r, T> Iterator for ParentsIter<'r, T> {
+    type Item = NodeRef<'r, T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let parent = self.node.parent()?;
```
Comment (gmanev7) [inline packages/tree/src/node.rs:109]: this is unstable yet https://doc.rust-lang.org/src/core/option.rs.html#2285

Comment (gmanev7) [reply packages/tree/src/node.rs:109]: how does it compile?

File: packages/tree/src/node.rs [hunk 2]
```diff
@@ -46,13 +51,59 @@ impl<'r, T> NodeRef<'r, T> {
...
+        Subtree::from_node(*self)
+    }
+
+    #[inline]
+    pub(crate) fn tree(&self) -> &'r Tree<T> {
+        self.tree
+    }
+
+    #[inline]
+    pub(crate) fn this_index(&self) -> u16 {
```
Comment (gmanev7) [inline packages/tree/src/node.rs:83]: u16 !!!

File: packages/tree/src/node.rs (hunk 1 above)
Comment (KirilMihaylov) [reply packages/tree/src/node.rs:109]: As discussed in DMs, this is just a sugar. It's usable with both Result and Option types.

**Pull request merged**